mdata/aggregator_test.go

```diff
@@ -91,7 +91,35 @@ func TestAggregator(t *testing.T) {
 	}
 	compare("simple-min-one-block", agg.minMetric, expected)
 
-	agg = NewAggregator(mockstore, &cache.MockCache{}, test.GetAMKey(2), ret, aggs, false)
+	// chunkspan is 120, ingestFrom = 140 means points before chunk starting at 240 are discarded
+	agg = NewAggregator(mockstore, &cache.MockCache{}, test.GetAMKey(1), ret, aggs, false, 140)
+	agg.Add(100, 123.4)
+	agg.Add(110, 5)
+	agg.Add(130, 130)
+	expected = []schema.Point{}
+	compare("simple-min-ingest-from-all-before-next-chunk", agg.minMetric, expected)
+
+	// chunkspan is 120, ingestFrom = 115 means points before chunk starting at 120 are discarded
+	agg = NewAggregator(mockstore, &cache.MockCache{}, test.GetAMKey(1), ret, aggs, false, 115)
+	agg.Add(100, 123.4)
+	agg.Add(110, 5)
+	agg.Add(130, 130)
+	expected = []schema.Point{
+		{Val: 5, Ts: 120},
+	}
+	compare("simple-min-ingest-from-one-in-next-chunk", agg.minMetric, expected)
+
+	// chunkspan is 120, ingestFrom = 120 means points before chunk starting at 120 are discarded
+	agg = NewAggregator(mockstore, &cache.MockCache{}, test.GetAMKey(1), ret, aggs, false, 120)
+	agg.Add(100, 123.4)
+	agg.Add(110, 5)
+	agg.Add(130, 130)
+	expected = []schema.Point{
+		{Val: 5, Ts: 120},
+	}
+	compare("simple-min-ingest-from-on-chunk-boundary", agg.minMetric, expected)
```

Comment:
too tired now, but tomorrow i'ld like to look into this one last thing, about the timestamps going 130->120 and how that may or should not affect things.

Reply:
> LGTM. just need to look at one last thing tomorrow.

Thank you! Your changes look great.